src/SWP/TemplateEngineBundle/Controller/ContainerController.php

```diff
@@ -0,0 +1,280 @@
+use SWP\TemplateEngineBundle\Model\ContainerData;
+
+class ContainerController extends FOSRestController
+{
+    /**
+     * Lists all registered containers.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists all registered containers",
+     *     statusCodes={
+     *         200="Returned on success."
+     *     }
+     * )
+     * @Route("/api/{version}/templates/containers/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_list_containers")
+     * @Method("GET")
+     */
+    public function listAction(Request $request)
+    {
+        $entityManager = $this->get('doctrine')->getManager();
+        $paginator = $this->get('knp_paginator');
+        $containers = $paginator->paginate($entityManager->getRepository('SWP\TemplateEngineBundle\Model\Container')->getAll());
+
+        if (count($containers) == 0) {
+            throw new NotFoundHttpException('Containers were not found.');
+        }
+
+        return $this->handleView(View::create($this->container->get('swp_pagination_rep')->createRepresentation($containers, $request), 200));
+    }
+
+    /**
+     * Get single container.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Get single container",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         404="Container not found",
+     *         422="Container id is not number"
+     *     }
+     * )
+     * @Route("/api/{version}/templates/containers/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_get_container")
+     * @Method("GET")
+     */
+    public function getAction(Request $request, $id)
+    {
+        if (!$id || !is_numeric($id)) {
+            throw new UnprocessableEntityHttpException('You need to provide container Id (integer).');
+        }
+
+        $entityManager = $this->get('doctrine')->getManager();
+        $container = $entityManager->getRepository('SWP\TemplateEngineBundle\Model\Container')->getById($id)->getOneOrNullResult();
+
+        if (!$container) {
+            throw new NotFoundHttpException('Container with this id was not found.');
+        }
+
+        return $this->handleView(View::create($container, 200));
+    }
+
+    /**
+     * Update single container.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Update single container",
+     *     statusCodes={
+     *         201="Returned on success.",
+     *         404="Container not found",
+     *         422="Container id is not number"
+     *     },
+     *     input="SWP\TemplateEngineBundle\Form\Type\ContainerType"
+     * )
+     * @Route("/api/{version}/templates/containers/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_update_container")
+     * @Method("PATCH")
+     */
+    public function updateAction(Request $request, $id)
+    {
+        if (!$id || !is_numeric($id)) {
```

Comment:
is_numeric also allows for floats, if it should be an integer then add is_int($id);